### PR TITLE
Migrating files to TS in shared folder

### PR DESCRIPTION
**Pull request by raztz15**

### CODE REVIEW

File: src/shared/components/circleBadge/circleBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const CircleBadge = ({ count = 0 }) => {
+	return (
+		<aside className='inline-flex text-primary-color-text bg-primary-color-background text-sm rounded-full h-3 w-3 p-3 justify-center items-center font-light'>
+			{count}
+		</aside>
+	);
```
Comment (Shaybush) [inline src/shared/components/circleBadge/circleBadge.tsx]: missing interface here

File: src/shared/components/controllers/Controllers.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+// should send title options and set search set option
+// controllers of all the tables
```
Comment (Shaybush) [inline src/shared/components/controllers/Controllers.tsx:3]: delete comments

File: src/shared/components/image-preview/ImagePreview.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+import ExitFill from '../../../assets/icons/exitFill';
+
+interface IImagePreviewpROPSmODEL {
```
Comment (Shaybush) [inline src/shared/components/image-preview/ImagePreview.tsx]:
```suggestion
import ExitFill from '../../../assets/icons/exitFill';

interface IImagePreviewPropsModel {
```

File: src/shared/components/controllers/Controllers.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+interface IControllerProps {
+	title: string;
+	placeHolder?: string;
+	options: { name: string; value: string }[];
+	handleSearchChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+	handleOptionChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+}
```
Comment (Shaybush) [inline src/shared/components/controllers/Controllers.tsx]:
```suggestion
interface IControllerPropsModel {
	title: string;
	placeHolder?: string;
	options: { name: string; value: string }[];
	handleSearchChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
	handleOptionChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
}
```

File: src/shared/components/controllers/Controllers.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+const Controllers = ({
+	title,
+	placeHolder = '',
+	options,
+	handleSearchChange,
+	handleOptionChange,
+}: IControllerProps) => {
```
Comment (Shaybush) [inline src/shared/components/controllers/Controllers.tsx]:
```suggestion
	handleSearchChange,
	handleOptionChange,
}: IControllerPropsModel) => {
```

File: src/shared/components/image-preview/ImagePreview.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+	id: string;
+	src: string;
+	alt: string;
+	onDeleteImg: (imgId: string) => void;
+}
+
+export const ImagePreview = ({ id, src, alt, onDeleteImg }: IImagePreviewpROPSmODEL) => {
```
Comment (Shaybush) [inline src/shared/components/image-preview/ImagePreview.tsx]:
```suggestion
}

export const ImagePreview = ({ id, src, alt, onDeleteImg }: IImagePreviewPropsModel) => {
```

File: src/shared/components/loadingComponents/loadingSideBar/loadingSideBar.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+const LoadingSideBar = ({ barsCount = 1 }) => {
+	return (
+		<div
+			role='status'
+			className='px-2 pb-2 pt-6 mt-8 border border-gray-200 bg-white rounded shadow animate-pulse lg:w-2/12 z-10 top-16 -left-1 lg:fixed'
+		>
```
Comment (Shaybush) [inline src/shared/components/loadingComponents/loadingSideBar/loadingSideBar.tsx]: add interface ILoadingSideBarPropsModel
optional to just do like that -
{ barsCount = 1 } : { barsCount?:number } 
if u want to

File: src/shared/components/rentBarChart/rentBarChart.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+// eslint-disable-next-line import/extensions
+import { isArrayEmpty, randomSetOfColors } from '../../../util/functions';
+
+interface IRentBarChartPropsModel {
+	config: { name: string }[];
+	width?: string;
+	height?: string;
+	backgroundColor?: string;
+	activeLegend: boolean;
+	opacity: number;
```
Comment (Shaybush) [inline src/shared/components/rentBarChart/rentBarChart.tsx]:
```suggestion
	activeLegend?: boolean;
	opacity?: number;
```
add Optional Chaining (?)

File: src/shared/components/rentCard/rentCard.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
+interface IRentCardPropsModel {
+	children?: string;
+	styleClass: string;
+	isColored?: boolean;
+	bottomColor?: string;
+	bottomContent?: string;
```
Comment (Shaybush) [inline src/shared/components/rentCard/rentCard.tsx]:
```suggestion
	children: string;
	styleClass: string;
	isColored?: boolean;
	bottomColor?: string;
	bottomContent: string;
```
add Optional Chaining (?) where needed and avoid to use where not

File: src/shared/components/rentLineChart/rentLineChart.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+interface IRentLineChartPropsModel {
+	config: [];
+	width?: string;
+	height?: string;
+	backgroundColor?: string;
+	strokeWidth?: string;
+	activeLegend?: boolean;
+	strokeDasharray?: string;
+}
```
Comment (Shaybush) [inline src/shared/components/rentLineChart/rentLineChart.tsx]:
```suggestion
interface IRentLineChartPropsModel {
	config: IRentLineChartConfigModel[];
	width?: string;
	height?: string;
	backgroundColor?: string;
	strokeWidth?: string;
	activeLegend?: boolean;
	strokeDasharray?: string;
}

interface IRentLineChartConfigModel {
	name: string;
	[key: string]: number | string;
}
```

File: src/shared/components/spinner-button/spinnerButton.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
+import { Rings } from 'react-loader-spinner';
+
+interface ILoadingButtonPropsModel {
+	isLoading: boolean;
+	children: string;
+	height: number;
+	width: number;
+}
```
Comment (Shaybush) [inline src/shared/components/spinner-button/spinnerButton.tsx]:
```suggestion
interface ILoadingButtonPropsModel {
	isLoading: boolean;
	children: string;
	height?: number;
	width?: number;
}
where default value exists add Optional Chaining (?)  avoid to use where not
```

File: src/shared/components/circleBadge/circleBadge.tsx [hunk 12]
```diff
@@ -0,0 +1,9 @@
+const CircleBadge = ({ count = 0 }) => {
+	return (
+		<aside className='inline-flex text-primary-color-text bg-primary-color-background text-sm rounded-full h-3 w-3 p-3 justify-center items-center font-light'>
+			{count}
+		</aside>
```
Comment (Shaybush) [inline src/shared/components/circleBadge/circleBadge.tsx]: add interface - ICircleBadgePropsModel

File: src/shared/components/loader/ballTriangle/ballTriangle.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { BallTriangle } from 'react-loader-spinner';
+
+interface IBallTriangleLoaderPropsModel {
+	load: boolean;
+	height: string;
+	width: string;
+	color: string;
```
Comment (Shaybush) [inline src/shared/components/loader/ballTriangle/ballTriangle.tsx]:
```suggestion
	load?: boolean;
	height: string;
	width: string;
	color?: string;
```
add Optional Chaining (?)

File: src/shared/components/circleBadge/circleBadge.tsx (hunk 1 above)
Comment (raztz15) [reply src/shared/components/circleBadge/circleBadge.tsx]: Done!

File: src/shared/components/controllers/Controllers.tsx (hunk 2 above)
Comment (raztz15) [reply src/shared/components/controllers/Controllers.tsx:3]: These comments are there for explanation of the component itself, are you sure you want to delete them?

File: src/shared/components/loadingComponents/loadingSideBar/loadingSideBar.tsx (hunk 7 above)
Comment (raztz15) [reply src/shared/components/loadingComponents/loadingSideBar/loadingSideBar.tsx]: Done

File: src/shared/components/rentBarChart/rentBarChart.tsx (hunk 8 above)
Comment (raztz15) [reply src/shared/components/rentBarChart/rentBarChart.tsx]: Done

File: src/shared/components/rentCard/rentCard.tsx (hunk 9 above)
Comment (raztz15) [reply src/shared/components/rentCard/rentCard.tsx]: Done

File: src/shared/components/rentLineChart/rentLineChart.tsx (hunk 10 above)
Comment (raztz15) [reply src/shared/components/rentLineChart/rentLineChart.tsx]: Done

File: src/shared/components/spinner-button/spinnerButton.tsx (hunk 11 above)
Comment (raztz15) [reply src/shared/components/spinner-button/spinnerButton.tsx]: Done

File: src/shared/components/circleBadge/circleBadge.tsx (hunk 12 above)
Comment (raztz15) [reply src/shared/components/circleBadge/circleBadge.tsx]: Done

File: src/shared/components/loader/ballTriangle/ballTriangle.tsx (hunk 13 above)
Comment (raztz15) [reply src/shared/components/loader/ballTriangle/ballTriangle.tsx]: Done

**Pull request merged**